src/diagonal.works/b6/api/functions/collections.go

```diff
@@ -341,3 +342,177 @@ var _ api.Collection = &flattenCollection{}
 func flatten(_ *api.Context, c api.Collection) (api.Collection, error) {
 	return &flattenCollection{c: c}, nil
 }
+
+type bound struct {
+	lower, upper interface{}
+	prettyPrint  interface{}
+	within       func(interface{}) (bool, error)
+}
+type buckets []bound
+
+func xBound(lower, upper interface{}) bound {
```

Comment:
reluctant to type new but i assume i have to :( miss cpp defaults, all of this would be a bit nicer i think..